instruments/masterMidiInterface.orc

```diff
@@ -0,0 +1,143 @@
+; masterMidiInterface
+
+/****************************
+ MIDI Status Codes:
+ 128 (note off)
+ 144 (note on)
+ 160 (polyphonic aftertouch)
+ 176 (control change)
+ 192 (program change)
+ 208 (channel aftertouch)
+ 224 (pitch bend
+ 0 if no MIDI message are pending in the MIDI IN buffer
+ ****************************/
+
+massign 1, "MasterMidiInterface"
+alwayson "MasterMidiInterface"
+
+instr MasterMidiInterface
+	kstatus, kchan, kdata1, kdata2  midiin 
+	ktrigger  changed  kstatus, kchan, kdata1, kdata2
+
+	if ktrigger ==1 && (kstatus == 0 || kstatus == 128) then
+		kgoto end
+	endif
+
+	if ktrigger==1 then
+	; -- print midi data to the terminal with formatting --
+	 printks "status:%d%tchannel:%d%tdata1:%d%tdata2:%d%n"\
+	                                    ,0,kstatus,kchan,kdata1,kdata2
+	endif
+	
+	if  kdata2 != 0 && kstatus == 144 then
+		kvelocity = (kdata2 / 127) ;TO DO: Get this into p6 of the scoreline statementes below
+
+		/****************************
+		 Drum Kit Notes
+		 ****************************/
+		; Ride
+		if kdata1 == 49  then
```

Comment:
I hate this series of if statements. I would like to find a better solution but my lack of syntax knowledge is holding me back. I would love some suggestions here.

Reply:
Looks like due to Csounds syntax, there is no "switch" statement included? http://csound.1045644.n5.nabble.com/idea-switch-case-statements-td5744547.html This is 3 yrs old though...